server/controllers/booking/booking.controller.js

```diff
@@ -43,13 +50,66 @@ const createBooking = function(req, res) {
             console.log(validateErrs);
             res.status(500).send(errs);
           } else {
-            newBooking.save((err, booking) => {
-              if (err) {
-                res.status(504).send(err);
-              } else {
-                res.json(booking._id);
-              }
-            });
+            Booking.find({ car: req.body.car, disabled: false })
```

Comment:
You don't need to check bookings for current users - only bookings for this car. You can add times as parameters for the request like here https://github.com/carshareadm/carshare/blob/master/server/controllers/cars/cars.controller.js#L60 and check if it returns null (no bookings) then you won't need that many checks

Reply:
Updated